download_data.py

```diff
@@ -155,23 +177,31 @@ def download_data(dataset,
             file_names = [f"{video_id}.zip", ]
         else:
             raise Exception(f'No such dataset = {dataset}')
-        os.makedirs(dst_dir, exist_ok=True)
 
         if should_download_laser_scanner_point_cloud and dataset == 'raw':
             # Point clouds only available for the raw dataset
             download_laser_scanner_point_clouds_for_video(video_id, metadata, download_dir)
 
         for file_name in file_names:
-            dst_zip = os.path.join(dst_dir, file_name)
+            dst_path = os.path.join(dst_dir, file_name)
             url = url_prefix.format(file_name)
 
-            if download_file(url, file_name, dst_dir):
-                # unzipping data
-                if file_name.endswith('.zip'):
-                    command = f"unzip -o {dst_zip} -d {dst_dir}"
-                    subprocess.check_call(command, shell=True)
-                    if not keep_zip:
-                        os.remove(dst_zip)
+            if not file_name.endswith('.zip'):
+                # target is not zip file
+                download_file(url, file_name, dst_dir)
+            else:
+                # target is zip file
+                if not os.path.isdir(dst_path[:-4]) and not os.path.isfile(dst_path):
+                    # zip file and unzip folder do not exist => download + unzip
+                    if download_file(url, file_name, dst_dir):
+                        unzip_file(file_name, dst_dir, keep_zip)
+                elif not os.path.isdir(dst_path[:-4]) and os.path.isfile(dst_path):
+                    # zip file exist but unzip folder do not exist => unzip only
+                    print(f'WARNING: skipping download of existing zip file: {dst_path}')
+                    unzip_file(file_name, dst_dir, keep_zip)
+                else:
+                    # unzip folder exist
+                    print(f'WARNING: skipping download and unzip of target zip file: {dst_path}')
```

Comment:
Consider the following adaptation to this code:
``` 
if download_file(url, dst_path, dst_dir):
    if file_name.endswith('.zip') and not os.path.isdir(dst_path[:len('.zip')]):
        unzip_file(file_name, dst_dir, keep_zip)
```

Reply:
I change the unzip process and remove the mechanism of unzipping to a temp file. Then we have:
1. If unzip folder does not exist, we can run function *down_file* no matter whether the target zip file exists.
2. No matter what, unzip the zip file if it exists. Since the cost of unzipping a file is low.
3. if unzip folder exist and the zip file does not exist, we can be sure that the file was decompressed normally.

The new logic might be a bit slower than the previous logic I designed before, but the code might be cleaner.